pkg/clusteragent/languagedetection/patcher.go

```diff
@@ -191,10 +196,30 @@ func (lp *languagePatcher) handleDeploymentEvent(event workloadmeta.Event) {
 		namespace,
 	)
 
-	if len(annotationsPatch) > 0 {
-		lp.patchOwner(&owner, annotationsPatch)
-	} else {
+	if event.Type == workloadmeta.EventTypeUnset {
+		// In case of unset event, we should clear language detection annotations if they are still present
+		// If they aren't present, then the resource has been deleted, so we should skip
+
+		if len(deployment.InjectableLanguages) > 0 {
+			// If some annotations still exist, remove them
+			annotationsPatch := lp.generateAnnotationsPatch(deployment.InjectableLanguages, langUtil.ContainersLanguages{})
+			lp.patchOwner(&owner, annotationsPatch)
+			return
+		}
+
 		SkippedPatches.Inc(owner.Kind, owner.Name, owner.Namespace)
+	} else if event.Type == workloadmeta.EventTypeSet {
+		detectedLanguages := deployment.DetectedLanguages
+		injectableLanguages := deployment.InjectableLanguages
+
+		// Calculate annotations patch
+		annotationsPatch := lp.generateAnnotationsPatch(injectableLanguages, detectedLanguages)
+		if len(annotationsPatch) > 0 {
+			lp.patchOwner(&owner, annotationsPatch)
+		} else {
+			SkippedPatches.Inc(owner.Kind, owner.Name, owner.Namespace)
```

Comment:
An increase in this metric indicates that the patcher is receiving unnecessary events while there is no need to patch deployments. 

It can be useful for performance evaluation and for debugging.